pkgs/by-name/ve/veloren/package.nix

```diff
@@ -0,0 +1,127 @@
+{ lib
+, rustPlatform
+, fetchFromGitLab
+, runCommandLocal
+, pkg-config
+, vulkan-loader
+, alsa-lib
+, udev
+, shaderc
+, xorg
+, libxkbcommon
+}:
+
+let
+  # Note: use this to get the release metadata
+  # https://gitlab.com/api/v4/projects/10174980/repository/tags/v{version}
+  version = "0.15.0";
+  date = "2023-06-30-15:09";
+  rev = "75c3b8ca659b6b740ebea8db46d2ebccb5bac6c6";
+
+  src = fetchFromGitLab {
+    owner = "veloren";
+    repo = "veloren";
+    inherit rev;
+    hash = "sha256-TKs/WT5S1hUBd9Y7AivFwbhlMI92GA5Auaevrjnx1cE=";
+  };
+
+  # Game assets, should not to be cached by hydra
+  assets = lib.dontDistribute (runCommandLocal "veloren-assets" {} ''
+    mkdir -p "$out/share/veloren"
+    cp -r '${src}/assets' "$out/share/veloren"
+  '');
+in
+
+rustPlatform.buildRustPackage {
+  pname = "veloren";
+  inherit src version;
+
+  cargoLock.lockFile = ./Cargo.lock;
```

Comment:
Is there a reason we can't use the upstream lockfile? Using it might make this package easier to maintain, as well as [avoiding the undesired side effects listed here](https://github.com/NixOS/nixpkgs/issues/327064).

```nix
cargoLock.lockFile = src + "/Cargo.lock";
```

Reply:
No idea, probably I just didn't think of this.

Reply:
It seems to work.

Reply:
This is input from derivation, and isn't allowed on ofborg and hydra.

Reply:
Here's what the [manual](https://nixos.org/manual/nixpkgs/stable/#compiling-rust-applications-with-cargo) says:

> Exception: If the application has cargo git dependencies, the cargoHash/cargoSha256 approach will not work, and you will need to copy the Cargo.lock file of the application to nixpkgs and continue with the next section for specifying the options of the cargoLock section.

So, that's why I had to include the lock file.